pkg/controller/s3/bucket/accelerateConfig.go

```diff
@@ -68,7 +69,7 @@ func NewAccelerateConfigurationClient(client s3.BucketClient) *AccelerateConfigu
 // Observe checks if the resource exists and if it matches the local configuration
 func (in *AccelerateConfigurationClient) Observe(ctx context.Context, bucket *v1beta1.Bucket) (ResourceStatus, error) {
 	external, err := in.client.GetBucketAccelerateConfigurationRequest(&awss3.GetBucketAccelerateConfigurationInput{Bucket: aws.String(meta.GetExternalName(bucket))}).Send(ctx)
-	if err != nil {
+	if err != nil || strings.Contains(err.Error(), "MethodNotAllowed") || strings.Contains(err.Error(), "UnsupportedArgument") {
```

Comment:
For classifying errors, we use a specific type of function so that it can be used in our other helper functions like `resource.Ignore`. You can see some examples [here](https://github.com/crossplane/provider-aws/blob/b935c3e/pkg/clients/s3/bucket.go#L153).

Reply:
Additionally, are we certain that `UnsupportedArgument` happens if and only if the accelerate config is not available in the region? It feels like we could get the same error for having unnecessary additional fields in the HTTP request, which isn't really possible since SDK designates it, but I wanted to know if there is a documentation somewhere that we can be sure about it. Also a comment here in the code would definitely help future readers.

Reply:
Yes there is! In the linked issue, there is a link to how the Terraform provider for aws handles it. I lifted this method from there cause I had the same concerns.